src/chain.h

```diff
@@ -317,7 +294,7 @@ class CBlockIndex
 
     //! Raise the validity level of this block index entry.
     //! Returns true if the validity was changed.
-    bool RaiseValidity(enum BlockStatus nUpTo)
+    bool RaiseValidity(enum BlockStatus nUpTo) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
faa2cdc58204823fb275905870f7dfee337f80ab

Move this to validation.cpp instead?
```cpp
static bool RaiseValidity(const CBlockIndex* pindex, enum BlockStatus nUpTo) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Reply:
I'll leave it here for now, because someone might want to make `nStatus` private at some point.

Reply:
I think it's preferable to deal with that later and avoid cs_main here now.

Reply:
This is under cs_main one way or another. You mean it is preferable to not include the sync header here?

Reply:
I mean this is only used in validation.cpp. Also, this kind of adds a circular dependency chain.h <-> validation.h (it's a forward declaration but still). I also don't think "this might be private some day" is valid argument. And by moving to validation.cpp you don't need 
to include <sync.h>.

Reply:
Removed commit instead. This can be addressed later.